README.md

```diff
@@ -28,17 +28,61 @@ Simple data loader for CGP HCA Data Store
 Because this program uses Amazon Web Services and Google Cloud Platform, you will need to set up credentials
 for both of these before you can run the program.
 
-### AWS credentials
+### AWS Credentials
 1. If you haven't already you will need to make an IAM user and create a new access key. Instructions are
    [here](https://docs.aws.amazon.com/general/latest/gr/managing-aws-access-keys.html).
 
 1. Next you will need to store your credentials so that Boto can access them. Instructions are
    [here](https://boto3.readthedocs.io/en/latest/guide/configuration.html).
 
-### GCP credentials
+### GCP Credentials
 1. Follow the steps [here](https://cloud.google.com/docs/authentication/getting-started) to set up your Google
    Credentials.
 
+## (Optional) Cloud Metadata Credentials Setup
+When the loader submits data, it actually needs access to the referenced files in the account to obtain metadata 
+(e.g. hash and size) that may be in another account.
+
+If the data is public, this is unnecessary.  However, if it is private data, a second account that is authorized 
+needs to be specified.
+
+### (Optional) GCP Metadata Credentials
+Make sure you have [gcloud](https://cloud.google.com/sdk/install) installed and run:
+
+1. gcloud auth application-default login
+
+1. Follow the link to the account accessed.
+
+1. This will generate a json with your user credentials with a path similar to:
+
+    `/home/<user>/.config/gcloud/application_default_credentials.json`
+
+1. Copy this json to another location so that it will not accidentally be used as a default by the main application.
+
+1. This file can then be used by the loader by specifying (as an example):
+
+    `--gce-metadata-cred=/home/<user>/metadata_credentials/application_default_credentials.json`
```

Comment:
Rather than `application_default_credentials.json` consider `my_user_credentials.json`.
This suggests naming the file according to the Google user whose credentials the file contains.
I don't think there is a way to know what user the credentials are for aside from the name of the file.